tests/kernel/common/src/irq_offload.c

```diff
@@ -199,3 +199,46 @@ __no_optimization void test_nop(void)
 	zassert_true(diff > 0,
 			"arch_nop() takes %d cpu cycles", diff);
 }
+
+static struct k_timer nestoff_timer;
+static bool timer_executed, nested_executed;
+
+void nestoff_offload(const void *parameter)
+{
+	nested_executed = true;
+}
+
+
+static void nestoff_timer_fn(struct k_timer *timer)
+{
+	zassert_false(nested_executed, "nested irq_offload ran too soon");
+	irq_offload(nestoff_offload, NULL);
+	zassert_true(nested_executed, "nested irq_offload did not run");
+
+	/* Set this last, to be sure we return to this context and not
+	 * the enclosing interrupt
+	 */
+	timer_executed = true;
+}
+
+
+/* Invoke irq_offload() from an interrupt and verify that the
+ * resulting nested interrupt doesn't explode
+ */
+void test_nested_irq_offload(void)
+{
+	if (!IS_ENABLED(CONFIG_IRQ_OFFLOAD_NESTED)) {
+		ztest_test_skip();
+	}
+
+	k_timer_init(&nestoff_timer, nestoff_timer_fn, NULL);
+
+	zassert_false(timer_executed, "timer ran too soon");
+	zassert_false(nested_executed, "nested irq_offload ran too soon");
+
+	k_timer_start(&nestoff_timer, K_MSEC(1000), K_FOREVER);
```

Comment:
Whoops, this was supposed to be K_TICKS(1), I bumped the time to be able to watch the logging and forgot to tune it back down before pushing.  Will fix.